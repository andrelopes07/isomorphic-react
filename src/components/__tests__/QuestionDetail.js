import React from 'react';
import renderer from 'react-test-renderer';
import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail';

describe('The Question Detail Component', () => {
    describe('The Container Element', () => {
        describe('mapStateToProps', () => {
            it('Should map state to props correctly', () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: 'Some body'
                };
                const appState = {
                    questions: [sampleQuestion]
                };
                const ownProps = {
                    question_id: 42
                };
                const componentState = mapStateToProps(appState, ownProps);

                expect(componentState).toEqual(sampleQuestion);
            });
        });
    });

    describe('mapStateToProps', () => {
        it(`Should not regress`, () => {
            const component = renderer.create(
                <QuestionDetailDisplay
                    title='Some Title'
                    body='Some Body'
                    answer_count={0}
                    tags={['tag1', 'tag2']} />
            );
            expect(component.toJSON()).toMatchSnapshot();
        });    
    });
});
