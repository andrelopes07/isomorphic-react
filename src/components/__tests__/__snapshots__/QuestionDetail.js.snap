// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component mapStateToProps Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Some Title
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          tag1
        </code>
        <code>
          tag2
        </code>
      </div>
    </div>
    <div>
      <p>
        Some Body
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
